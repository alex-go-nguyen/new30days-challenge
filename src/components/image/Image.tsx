import * as React from 'react';
import classNames from 'classnames/bind';
import styles from './Image.module.scss';

const cx = classNames.bind(styles);

type ImageBorderRadius = 'none' | 'little' | 'partial' | 'full';

export interface ImageProps {
    ratio?: number;
    src: string;
    alt?: string;
    borderRadius?: ImageBorderRadius;
}

export default function Image({ ratio = 4 / 3, src, alt = 'Squirrel', borderRadius = 'none' }: ImageProps) {
    return (
        <div className={cx('container', borderRadius)}>
            <style jsx>
                {`
                    div {
                        aspect-ratio: ${ratio};
                    }
                `}
            </style>
            <img src={src} alt={alt} className={cx('image')} />
        </div>
    );
}
