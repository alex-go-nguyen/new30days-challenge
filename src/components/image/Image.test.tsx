import renderer from 'react-test-renderer';
import React from 'react';
import Image from './Image';

describe('Image component', () => {
    const component = renderer.create(<Image src={'./cold.png'} />);
    let tree = component.toJSON();
    test('should render correctly', () => {
        expect(tree).toMatchSnapshot();
    });
});
