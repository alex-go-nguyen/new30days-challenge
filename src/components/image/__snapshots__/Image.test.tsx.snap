// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image component should render correctly 1`] = `
<div
  className="container none"
>
  <style
    jsx={true}
  >
    
                    div {
                        aspect-ratio: 1.3333333333333333;
                    }
                
  </style>
  <img
    className="image"
    src="./cold.png"
  />
</div>
`;
